### ISSUE TITLE
Combined Date/Time Selector

### ISSUE THREAD

**Issue by Kyle-Cooley** [IBM/carbon-components-react#1140]
## Detailed description

> Describe in detail the issue you're having.

For the project I am working on, we need a combined dropdown selector for date and time. As far as I can tell, this does not appear to exist in carbon. This seems like something that could be useful for multiple projects. Just wondering if there was an interest in adding/amending a component for this use case.

> Is this a feature request (new component, new icon), a bug, or a general issue?

Feature request

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

APM

## Additional information

The DatePicker component could enable this, since flatpickr has properties for this. They just need to be exposed to the users of carbon. There are a couple of css issues that would need resolved, but it is an option. I'm just not sure whether or not it would fit with the carbon design.

![image](https://imgur.com/HEByquy.png)

**Comment by asudoh** [comment]
@aagonzales @designertyler Is this usage pattern something we can allow? Thanks!

**Comment by aagonzales** [comment]
Yeah I think this could be a good pattern. My only concern/feedback is figuring out how to show the selected time in the closed view so its still visible to user when closed. Maybe inline or stacked with the date?

**Comment by ShuangLiLi** [comment]
@aagonzales We want a range date picker with time selection, how about the schedule? We are urgent about the new version or workaround. Thanks!

**Comment by asudoh** [comment]
Hi @ShuangLiLi though I don't think this feature is in our imminent roadmap due to the number of teams asking for this - You can create a custom React component with Carbon style and some custom styles (and possibly Flatpickr our component libraries use). PRs are appreciated if you come up with a component that aligns to Carbon design well and is solid and flexible enough for many teams to use. Thanks!

**Comment by ShuangLiLi** [comment]
@aagonzales @asudoh  Hi, I also use flatpickr to support range date picker can select time, but how can I show the time in label with date?

Also, I have another question about how to set a default date and time for a DatePickerInput?

Thanks a lot for your help!

**Comment by diego-codes** [comment]
Just chiming in, my application also requires this type of solution. It would be great if Carbon provided this out of the box. I will do the workaround that @asudoh suggested, but I would urge you to making this feature a higher priority 😅.

**Comment by martynjames** [comment]
My application already has this kind of component, and we'd love to be able to switch to using carbon

**Comment by aagonzales** [comment]
@diego-codes or @martynjames if your teams are building a version it would be good to have them contribute it back to the main Carbon library so other teams can use it as well. The core Carbon team has limited resources which is why we unfortunately can't get to some of these new features.

**Comment by asudoh** [comment]
Given I think we are in "help wanted" mode, I marked it as such. Also marked the style change (#1586) as "ready for review" to see what our designers think of the latest state of the contribution, though I think more work than enabling Flatpickr's "date/time combined" is needed. Thanks!
